### DISCUSSION TITLE
eglot+jdtls gives me: no definitions found for lsp identifier at point

### DISCUSSION THREAD

**Discussion by cabins** [joaotavora/eglot#842] [Q&A] [answered]
I use Eglot + jdtls for Java codes, when I press `M-.` for definitions I got the error:

```plain
No definitions found for: LSP identifier at point.
```

and the code did NOT jump to the expected source code. The communications between eglot and jdtls:
![image](https://user-images.githubusercontent.com/2500123/155253611-c8917ca1-0ac8-4de9-bc63-c32f57167020.png)

Then I compare the `lsp-mode` for the same code and same jdtls, I got the communications like this:
![image](https://user-images.githubusercontent.com/2500123/155253655-c7a82519-250a-4e83-8d66-95f3c9314324.png)

and `lsp-mode` jumps to the expected source code.

Anyone can tell me what happened to me? Thanks!!

**Comment by nemethf** [comment, marked as answer]
It the first case you start from character 46, in the second case you start from 47.  Are there non-ascii characters involved? 

Are the servers configured with the same options in both cases?

Have you configured eglot-extend-to-xref?  (Currently it doesn't matter, the server doesn't return a reference.)

I'm also guessing that tramp is able to open jdt:// files.  I'm not sure Eglot works with non- file:// URIs.  It probably does.

**Reply by cabins** [reply]
https://emacs-lsp.github.io/lsp-mode/page/remote/

Yes!! I think you are right! Please accept all my thanks!

**Comment by theothornhill** [comment]
Sorry to bring this back up, but @cabins, did you ever get eglot to return anything? I struggle with the same issue. Eglot returns empty array, lsp-mode and vscode returns a jdt:// prefixed file.

I don't see how eglot-extend-to-xref would change anything there, @nemethf?
